packages/fe-fpm-writer/src/filter/index.ts

```diff
@@ -0,0 +1,103 @@
+ */
+function enhanceConfig(data: CustomFilter, manifestPath: string, manifest: Manifest): InternalCustomFilter {
+    // clone input
+    const config: CustomFilter & Partial<InternalCustomFilter> = data;
+
+    setCommonDefaults(config, manifestPath, manifest);
+
+    // set default values for requirement, language, the fragment file name
+    config.required = config.required || false;
+    config.typescript = !!config.typescript;
+    config.fragmentFile = config.fragmentFile || config.name;
+
+    return config as InternalCustomFilter;
+}
+
+/**
+ * Add a custom filter to the filter bar of an existing UI5 application.
+ *
+ * @param {string} basePath - the base path
+ * @param {CustomFilter} filterConfig - the custom filter configuration
+ * @param {Editor} [fs] - the memfs editor instance
+ * @returns {Promise<Editor>} the updated memfs editor instance
+ */
+export function generateCustomFilter(basePath: string, filterConfig: CustomFilter, fs?: Editor): Editor {
+    if (!fs) {
+        fs = create(createStorage());
+    }
+    validateBasePath(basePath, fs);
+
+    const manifestPath = join(basePath, 'webapp/manifest.json');
+    const manifest = fs.readJSON(manifestPath) as Manifest;
+    const config = enhanceConfig(filterConfig, manifestPath, manifest);
+
+    // Apply event handler
+    if (config.eventHandler) {
+        config.eventHandler = applyEventHandlerConfiguration(
+            fs,
+            config,
+            config.eventHandler,
+            false,
+            config.typescript,
+            contextParameter,
+            { templatePath: 'filter/Controller', fnName: 'itemsFilter', fileName: config.name }
+        );
+    } else {
+        // create a controller file
+        const ext = filterConfig.typescript ? 'ts' : 'js';
+        const viewPath = join(config.path, `${config.name}.${ext}`);
+        if (!fs.exists(viewPath)) {
+            fs.copyTpl(getTemplatePath(`filter/Controller.${ext}`), viewPath, config);
+        }
+    }
+
+    // enhance manifest with the filter definition and controller reference
+    const filters = enhanceManifestAndGetFiltersReference(manifest, filterConfig);
+    Object.assign(filters, JSON.parse(render(fs.read(getTemplatePath(`filter/manifest.json`)), config, {})));
+    fs.writeJSON(manifestPath, manifest, undefined, getJsonSpace(fs, manifestPath, filterConfig.tabInfo));
+
+    // create a fragment file
+    const fragmentPath = join(config.path, `${config.fragmentFile}.fragment.xml`);
+    fs.copyTpl(getTemplatePath(`filter/fragment.xml`), fragmentPath, config);
+
+    return fs;
+}
+
+/**
+ * Enhance the target object in the manifest with the required nested objects and return a reference to it.
+ *
+ * @param {Manifest} manifest - the application manifest
+ * @param {CustomFilter} customFilter - the custom filter configuration
+ * @returns Filters object of the given page
+ */
+function enhanceManifestAndGetFiltersReference(manifest: any, customFilter: CustomFilter): any {
```

Comment:
You have another `any` in line 98. Please ensure that you do not use any. If you don't know the type `unknown` is the correct choice because it forces you to handle the variable like a variable you don't anything about.